com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Camera/HDCameraUI.Drawers.cs

```diff
@@ -298,14 +303,33 @@ static void Drawer_PhysicalCamera(SerializedHDCamera p, Editor owner)
             using (new EditorGUI.IndentLevelScope())
             {
                 EditorGUI.BeginChangeCheck();
-                int filmGateIndex = Array.IndexOf(k_ApertureFormatValues, new Vector2((float)Math.Round(cam.sensorSize.vector2Value.x, 3), (float)Math.Round(cam.sensorSize.vector2Value.y, 3)));
-                if (filmGateIndex == -1)
-                    filmGateIndex = EditorGUILayout.Popup(cameraTypeContent, k_ApertureFormatNames.Length - 1, k_ApertureFormatNames);
-                else
-                    filmGateIndex = EditorGUILayout.Popup(cameraTypeContent, filmGateIndex, k_ApertureFormatNames);
 
-                if (EditorGUI.EndChangeCheck() && filmGateIndex < k_ApertureFormatValues.Length)
-                    cam.sensorSize.vector2Value = k_ApertureFormatValues[filmGateIndex];
+                int oldFilmGateIndex = Array.IndexOf(k_ApertureFormatValues, new Vector2((float)Math.Round(cam.sensorSize.vector2Value.x, 3), (float)Math.Round(cam.sensorSize.vector2Value.y, 3)));
+
+                // If it is not one of the preset sizes, set it to custom
+                oldFilmGateIndex = (oldFilmGateIndex == -1) ? k_CustomPresetIndex: oldFilmGateIndex;
+
+                // Get the new user selection
+                int newFilmGateIndex = EditorGUILayout.Popup(cameraTypeContent, oldFilmGateIndex, k_ApertureFormatNames);
+
+                if (EditorGUI.EndChangeCheck())
+                {
+                    // When switching from custom to a preset, save the last custom value (to display again, in case the user switches back to custom)
+                    if (oldFilmGateIndex == k_CustomPresetIndex)
+                    {
+                        s_LastCustomSensorSizeValue = cam.sensorSize.vector2Value;
+                    }
+
+                    if (newFilmGateIndex < k_ApertureFormatValues.Length)
```

Comment:
Not a problem with this PR, but took me a second to realize that k_ApertureFormatValues is smaller than k_ApertureFormatNames which is somewhat confusing :(

Reply:
I already had a  k_CustomPresetIndex variable, so I used this one. 
 
Interestingly my first implementation was to expand the k_ApertureFormatValues by one and use the last place for the custom value storage, but in the end I think it is better to keep the array as read-only (and also keep changed lines to minimum).